src/createElements.js

```diff
@@ -37,6 +41,10 @@ export default function createElements(
       return Component ? <Component {...match} /> : null;
     }
 
+    if (route.prerender) {
+      route.prerender({ ...readyState, match });
```

Comment:
should pass the environment as well in order to read the query yeah?

Can queries be read twice? what will happen in the ReadyStateRenderer below?